slack-cmd.c

```diff
@@ -138,6 +186,30 @@ void slack_cmd_register() {
 	id = purple_cmd_register("delete", "", PURPLE_CMD_P_PRPL, PURPLE_CMD_FLAG_IM | PURPLE_CMD_FLAG_CHAT | PURPLE_CMD_FLAG_PRPL_ONLY,
 			SLACK_PLUGIN_ID, cmd_delete, "delete: remove your last message", NULL);
 	commands = g_slist_prepend(commands, GUINT_TO_POINTER(id));
+
+	// Make sure the number of %s matches the printfs further down.
+	const char *thread_fmt = "%s thread-timestamp message:  Post messages in threads.";
+	GString *thread_help = g_string_new(NULL);
+
+	g_string_printf(thread_help, thread_fmt, "th");
+	id = purple_cmd_register("th", "s", PURPLE_CMD_P_PRPL, PURPLE_CMD_FLAG_IM | PURPLE_CMD_FLAG_CHAT | PURPLE_CMD_FLAG_PRPL_ONLY |
+			PURPLE_CMD_FLAG_ALLOW_WRONG_ARGS, SLACK_PLUGIN_ID, cmd_thread, thread_help->str, NULL);
+	commands = g_slist_prepend(commands, GUINT_TO_POINTER(id));
+
+	g_string_printf(thread_help, thread_fmt, "thread");
+	id = purple_cmd_register("thread", "s", PURPLE_CMD_P_PRPL, PURPLE_CMD_FLAG_IM | PURPLE_CMD_FLAG_CHAT | PURPLE_CMD_FLAG_PRPL_ONLY |
```

Comment:
I think the arguments to this should be "ws" (or maybe "wS" -- been meaning to sort this out for edit), which will simplify the parsing. You also shouldn't need WRONG_ARGS.

(I'm not entirely sold on having the shorter aliases, since there's generally tab completions, but would be good to at least avoid duplication.)

Reply:
> I think the arguments to this should be "ws" (or maybe "wS" -- been meaning to sort this out for edit), which will simplify the parsing. You also shouldn't need WRONG_ARGS.

Yes, this is a remnant of the thread context feature, which is still being used in my local fork. I'll check out how much work it is to change this.

> (I'm not entirely sold on having the shorter aliases, since there's generally tab completions, but would be good to at least avoid duplication.)

You're right, I did not think of this. I can remove the aliases if you want, since with completion they're not really necessary.